trip-info/firstScript.js

```diff
@@ -0,0 +1,149 @@
+
+const oneMonthAgo = () =>{
+    var d = (new Date());
+    d.setMonth(d.getMonth()-1);
+     return (d.toISOString());
+    
+}
+const convertArrayToObject = (array, key) =>{
+    const initialValue = {};
+    return array.reduce((obj, item) => {
+        return {
+        ...obj,
+        [item[key]]: item,
+        };
+    }, initialValue);
+};
+
+
+const call = (method, params) => {
+    return new Promise((resolve, reject) => {
+        api.call(method, params, resolve, reject);
+    })
+}
+
+const multiCall  = (calls) => {
+    return new Promise((resolve, reject) => {
+        api.call("ExecuteMultiCall", {
+            calls : calls
+        }, resolve);
+    }) 
+}
+
+const getDrivers =  async () => {
+    const drivers = await call("Get",{
+        "typeName": "User",
+        "resultsLimit": 10,
+        "search": {
+            "isDriver": true
+         }
+    });
+    return convertArrayToObject(drivers,"id");
+}
+
+const getDriversTrips = async (drivers) => {
+    const now = (new Date()).toISOString();
+    const tripCalls = drivers.map(driver => ({
+        "method": "Get",
+        "params": {
+            "typeName": "Trip",
+            "resultsLimit": 10,
+            "search": {
+                "userSearch":{
+                    "id": driver.id
+                },
+                "toDate": now,
+                "fromDate": oneMonthAgo(),
+                "includeOverlappedTrips":true 
+            }
+        }
+    }));
+    return await multiCall(tripCalls);
+};
+
+
+const getAddresses = async (trips)  =>{
+    const addressCalls = trips.map(driverTrip => ({
+        "method" : "GetAddresses",
+        "params" : {
+            "coordinates": driverTrip.map((trip) => ({"x": trip.stopPoint.x, "y": trip.stopPoint.y})),
+            "movingAddresses":false,
+        }
+    }))
+    
+    return await multiCall(addressCalls);
+}
+
+const getDevices = async (trips) => {
```

Comment:
The function name should also represent what is being passed for small functions

For example: this would be getTripDevices